src/index.d.ts

```diff
@@ -0,0 +1,825 @@
+        optSet(key: "ascii-chars" | "ascii_chars" | "AsciiChars", value: boolean): void;
+        /**
+         * join-classes
+         */
+        optGet(key: "join-classes" | "join_classes" | "JoinClasses"): boolean;
+        optSet(key: "join-classes" | "join_classes" | "JoinClasses", value: boolean): void;
+        /**
+         * join-styles
+         */
+        optGet(key: "join-styles" | "join_styles" | "JoinStyles"): boolean;
+        optSet(key: "join-styles" | "join_styles" | "JoinStyles", value: boolean): void;
+        /**
+         * escape-cdata
+         */
+        optGet(key: "escape-cdata" | "escape_cdata" | "EscapeCdata"): boolean;
+        optSet(key: "escape-cdata" | "escape_cdata" | "EscapeCdata", value: boolean): void;
+        /**
+         * language
+         */
+        optGet(key: "language" | "Language"): string;
+        optSet(key: "language" | "Language", value: string): void;
+        /**
+         * ncr
+         */
+        optGet(key: "ncr" | "Ncr"): boolean;
+        optSet(key: "ncr" | "Ncr", value: boolean): void;
+        /**
+         * output-bom
+         */
+        optGet(key: "output-bom" | "output_bom" | "OutputBom"): "no" | "yes" | "auto";
+        optSet(key: "output-bom" | "output_bom" | "OutputBom", value: "no" | "yes" | "auto"): void;
+        /**
+         * replace-color
+         */
+        optGet(key: "replace-color" | "replace_color" | "ReplaceColor"): boolean;
+        optSet(key: "replace-color" | "replace_color" | "ReplaceColor", value: boolean): void;
+        /**
+         * css-prefix
+         */
+        optGet(key: "css-prefix" | "css_prefix" | "CssPrefix"): string;
+        optSet(key: "css-prefix" | "css_prefix" | "CssPrefix", value: string): void;
+        /**
+         * new-inline-tags
+         */
+        optGet(key: "new-inline-tags" | "new_inline_tags" | "NewInlineTags"): string;
+        optSet(key: "new-inline-tags" | "new_inline_tags" | "NewInlineTags", value: string): void;
+        /**
+         * new-blocklevel-tags
+         */
+        optGet(key: "new-blocklevel-tags" | "new_blocklevel_tags" | "NewBlocklevelTags"): string;
+        optSet(key: "new-blocklevel-tags" | "new_blocklevel_tags" | "NewBlocklevelTags", value: string): void;
+        /**
+         * new-empty-tags
+         */
+        optGet(key: "new-empty-tags" | "new_empty_tags" | "NewEmptyTags"): string;
+        optSet(key: "new-empty-tags" | "new_empty_tags" | "NewEmptyTags", value: string): void;
+        /**
+         * new-pre-tags
+         */
+        optGet(key: "new-pre-tags" | "new_pre_tags" | "NewPreTags"): string;
+        optSet(key: "new-pre-tags" | "new_pre_tags" | "NewPreTags", value: string): void;
+        /**
+         * accessibility-check
+         */
+        optGet(key: "accessibility-check" | "accessibility_check" | "AccessibilityCheck"): 0 | 1 | 2 | 3;
+        optSet(key: "accessibility-check" | "accessibility_check" | "AccessibilityCheck", value: 0 | 1 | 2 | 3): void;
```

Comment:
The `optGet` part is incorrect: this version does return a string from the picklist. The `optSet` part is not exhaustive, as all of these integer-typed pick-value options accept integers or strings. Might be worth modeling.